index.js

```diff
@@ -470,21 +494,45 @@ const createFocusTrap = function (elements, userOptions) {
         return this;
       }
 
-      updateTabbableNodes();
-
       state.active = true;
       state.paused = false;
       state.nodeFocusedBeforeActivation = doc.activeElement;
 
-      const onActivate =
-        activateOptions && activateOptions.onActivate
-          ? activateOptions.onActivate
-          : config.onActivate;
-      if (onActivate) {
-        onActivate();
+      const getOption = (optionName) => {
+        return activateOptions && activateOptions[optionName]
+          ? activateOptions[optionName]
+          : config[optionName];
+      };
+
+      const onActivate = getOption('onActivate');
+      const onSuccessfulActivation = getOption('onSuccessfulActivation');
+      const checkCanActivate = getOption('checkCanActivate');
+
+      if (checkCanActivate) {
+        if (onActivate) {
+          onActivate();
+        }
+
+        state.containers.forEach((container) => {
```

Comment:
I'm uncertain about this.

It works for my use case but my use case only has a single focus-trap container. I'm not sure if this will work for multiple focus-traps.

Reply:
Based on what the multiple elements with multiple traps example looks like, it doesn't look like the multi-element trap is a practical use case for animated dialogs. You are only going to animate 1 dialog at a time.

I made a test demo based on the multiple elements with multiple traps demo just to see if it would explode and it seemed to handle it ok.